design/metadata-handling.md

```diff
@@ -630,16 +654,22 @@ object will be reconciled. There will be two cases:
   it. The reconciler will also verify that the required secrets exist. If they
   do not, they will be created.
 - if no Metal3Data exists for that *Metal3DataClaim*, then the
-  reconciler will create one and fill the respective field with the secret name.
+  reconciler will create one and fill the respective field with the secrets names.
+  It will set the `templateReference` to the value of the Metal3DataTemplate if set.
 
 To create a Metal3Data object, the *Metal3DataClaim* controller will select an
 index for that Metal3Machine. The selection happens by selecting the lowest
 available index that is not in use. To do that, the controller will list all
-existing Metal3Data object linked to this Metal3DataTemplate and to get the
-unavailable indexes. The indexes always start from 0 and increment by 1. The
-lowest available index is to be used next. The `dataNames` field contains the
-map of Metal3Machine to Metal3Data and the `indexes` contains the map of
-allocated indexes and claims.
+existing Metal3Data object linked to this Metal3DataTemplate and get the
+unavailable indexes. The Metal3Data objects are linked to a Metal3DataTemplate
+by three ways. Either they directly reference the template in the `template`
+field of the `spec`, or they have the same `templateReference` key as the
+template, or the template's `templateReference` matches the `name` of the
```

Comment:
If I understood correctly, the third way is to bridge the gap between the old and new implementations, right ? Maybe that should be specified here, that it's a workaround for the migration ?

Reply:
True, I can repharse it.

Reply:
Addressed